storage/rocksdb/ha_rocksdb.cc

```diff
@@ -11534,8 +11534,8 @@ void ha_rocksdb::get_auto_increment(ulonglong off, ulonglong inc,
       // with the current value).
       //
       // See above explanation for inc == 1 for why we use std::min.
-    } while (!auto_incr.compare_exchange_weak(last_val,
-                                              std::min(new_val + 1, max_val)));
+    } while (!auto_incr.compare_exchange_weak(
+        last_val, std::min(new_val, max_val - 1) + 1));
```

Comment:
Can we extend the comment above to explain why we do this -1/+1 business? The original std::min expression was probably copied from the inc == 1 case.